From 2e7407a26a0794051625d07a9cf465bbcb793405 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Wed, 22 Jan 2025 12:05:06 +0800
Subject: [PATCH 185/210] mtk: hostapd: remove interface count check from
 hostapd_is_mld_ap()

This check causes missing MLE issue in probe response of a single-link MLD
AP, so if a STA uses the info from a probe response, it would consider AP
as non-MLO and then connect with legacy connection.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 src/ap/ieee802_11.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index b77bcc0f1..7008429a2 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -4801,10 +4801,6 @@ bool hostapd_is_mld_ap(struct hostapd_data *hapd)
 	if (!hapd->conf->mld_ap)
 		return false;
 
-	if (!hapd->iface || !hapd->iface->interfaces ||
-	    hapd->iface->interfaces->count <= 1)
-		return false;
-
 	return true;
 }
 
-- 
2.45.2

